### DISCUSSION TITLE
How do you add new kernel releases to linux-tegra-4.9

### DISCUSSION THREAD

**Discussion by lfdmn** [OE4T/meta-tegra#1343] [General]
Hi,

@madisongh what are the steps you are taking to merge the NVidia repositories, fix the layout, apply the OE4T patches, etc, in order to bring a new kernel release branch to [linux-tegra-4.9](https://github.com/OE4T/linux-tegra-4.9).

I'm still hunting the RT kernel locks and would like to get to something as close to NVidia sources as possible for testing, in case some community patch would introduce the problem.

Thanks,

**Comment by madisongh** [comment]
I have a script to create the subtrees, but it's somewhat error-prone to use and has zero documentation.   If you look at the git history of each of the branches, find the commit labeled "Build system updates for merged nvidia tree," which should be just after the final "Import subtree" commit.  The final import pulls in the last of NVIDIA's sources unmodified.  The "Build system updates" commit has just the Kbuild changes to make the build work in the unified tree.  That would be a good starting point.  I've just pushed `l4t-r32.7.3-base` and `l4t-r32.7.4-base` branches to the 4.9 repo to make those final imports easier to find for the last couple of releases.

You'll probably need to cherry-pick over a handful of the later commits to deal with build issues, unless you use the exact version of make and the toolchain that NVIDIA used for their builds.  There are one or two commits that address build issues using the Yocto kernel configuration tools, too.

If you really need/want to start from scratch, let me know and I'll post the script somewhere.

**Reply by lfdmn** [reply]
Thanks! I'll start iterating on top of those base branches.

**Reply by 0m-ax** [reply]
Could you share the script that makes the tree?

**Reply by madisongh** [reply]
I've uploaded it to https://github.com/OE4T/kernel-unification-scripts

Have fun
